### PR TITLE
Remove usage of multi-types from the docs and added a page explaining type removal

### LINKED ISSUE #25401: Fix documentation to not use multiple types
There are still some places in the documentation that should be fixed to either use the new parent join facilities or should be fixed in general to not use multiple types:

 * `reference/aggregations/bucket/children-aggregation.asciidoc`
 * `reference/docs/index_.asciidoc`
 * `reference/indices/put-mapping.asciidoc`
 * `reference/mapping/dynamic/default-mapping.asciidoc`
 * `reference/mapping/fields/parent-field.asciidoc`
 * `reference/mapping/fields/type-field.asciidoc`
 * `reference/mapping.asciidoc`
 * `reference/query-dsl/parent-id-query.asciidoc`

some of them might even be removed. Relates to #24961

### PR DESCRIPTION
**Pull request by clintongormley**
Closes #25401

### CODE REVIEW

**Comment by s1monw** [comment]
LGTM for what it's worth

File: docs/reference/cat/nodes.asciidoc [hunk 1]
```diff
@@ -184,7 +184,7 @@ segments |1.4kb
 |`segments.version_map_memory` |`svmm`, `segmentsVersionMapMemory` |No
 |Memory used by version map |1.0kb
 |`segments.fixed_bitset_memory` |`sfbm`, `fixedBitsetMemory` |No
-|Memory used by fixed bit sets for nested object field types and type filters for types referred in _parent fields |1.0kb
+|Memory used by fixed bit sets for nested object field types and type filters for types referred in `join fields |1.0kb
```
Comment (jpountz) [inline docs/reference/cat/nodes.asciidoc]: missing "`"?

File: docs/reference/mapping/dynamic/default-mapping.asciidoc [hunk 2]
```diff
@@ -1,89 +1,14 @@
 [[default-mapping]]
 === `_default_` mapping
 
-The default mapping, which will be used as the base mapping for any new
-mapping types, can be customised by adding a mapping type with the name
+deprecated[6.0.0,See <<removal-of-types>>]
```
Comment (jpountz) [inline docs/reference/mapping/dynamic/default-mapping.asciidoc]: ++

**Review by jpountz — APPROVED** [review]
LGTM

**Comment by clintongormley** [comment]
thanks @jpountz for the review

**Pull request merged**